homeassistant/components/s3/backup.py

```diff
@@ -0,0 +1,263 @@
+"""Backup platform for the S3 integration."""
+
+from collections.abc import AsyncIterator, Callable, Coroutine
+import functools
+import json
+import logging
+from time import time
+from typing import Any, TypeVar, cast
+
+from botocore.exceptions import BotoCoreError
+
+from homeassistant.components.backup import (
+    AgentBackup,
+    BackupAgent,
+    BackupAgentError,
+    BackupNotFound,
+    suggested_filename,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from . import S3ConfigEntry
+from .const import CONF_BUCKET, DATA_BACKUP_AGENT_LISTENERS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+CACHE_TTL = 300
+
+# Type variable for the decorator return type
+T = TypeVar("T")
+
+
+def handle_boto_errors(
```

Comment:
```suggestion

def handle_boto_errors[T](
```
Python 3.12 syntax should be fine